genomic-medicine-sweden/NGPIris
# Novel search modes

## Contents

- Added optional NGPi password to credentials
- Added main to all CLI files
- Rewrote download and search to have multiple search modes
- Readded search mode NGPi
- Readded search mode NGPi-legacy
- Redefined search mode NGPr
- Redefined search mode None
- Integrated all functionality of hci_functions to either functions, or misc set utils
- Added function indices
- Propagated urllib3 fix
- Rewrote HCI file into HCIManager class
- Reconfigured HCIManager to work with current NGPi
- Reordered Readme

### This [update](https://semver.org/) is:

## Test Procedure

### Installation and initiation
* `git clone git@github.com:genomic-medicine-sweden/NGPIris.git`
* `cd NGPIris && git checkout <BRANCH>`
* `bash setup.sh`
* `source activate hpcenv`

### Tests
* `pytest tests/`
* Try using both search modes
* Try using all 4 download-search modes
* Try using indices utils function
* Verify no SSH licece error shows up
* Look at the Readme and say that tis useful

### Expected outcome:
* Pytest resolves without crashes
* Potential additional results

## Confirmations:
- [x] Code reviewed by @JPWerling
